### PR TITLE
codewhisperer: send security issue to q chat

### PR DESCRIPTION
**Pull request by ctlai95**
## Problem

Integrate security scans with Amazon Q chat

## Solution

- Add "Explain with Q" and "Fix with Q" links in security issue hover, code actions, and webview
- Update chat controller `explainCode` and `fixCode` command to accept `CodeScanIssue` as argument

image
image
image
image

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhispererChat/commands/registerCommands.ts [hunk 1]
```diff
@@ -69,3 +71,8 @@ export interface EditorContextCommand {
     type: EditorContextCommandType
     triggerType: EditorContextCommandTriggerType
 }
+
+export interface EditorContextCommandWithIssue extends EditorContextCommand {
+    type: 'aws.amazonq.fixCode'
+    codeScanIssue: CodeScanIssue
+}
```
Comment (AnirudhKonduru) [inline packages/core/src/codewhispererChat/commands/registerCommands.ts:78]: why extend an existing command instead of just defining a new one?

**Review by ashishrp-aws — APPROVED** [review]

File: packages/core/src/codewhispererChat/commands/registerCommands.ts [hunk 2]
```diff
@@ -62,10 +81,20 @@ export type EditorContextCommandType =
...
+    | 'aws.amazonq.explainIssue'
+    | 'aws.amazonq.fixIssue'
 
 export type EditorContextCommandTriggerType = 'contextMenu' | 'keybinding' | 'commandPalette' | 'click'
 
-export interface EditorContextCommand {
-    type: EditorContextCommandType
+export interface EditorContextCommandBase {
+    type: Exclude<EditorContextCommandType, 'aws.amazonq.explainIssue' | 'aws.amazonq.fixIssue'>
     triggerType: EditorContextCommandTriggerType
```
[the remark this answers is not in the document]
Comment (ctlai95) [reply packages/core/src/codewhispererChat/commands/registerCommands.ts:91]: Sure will separate them out into two types

File: packages/core/src/codewhispererChat/commands/registerCommands.ts (hunk 1 above)
Comment (justinmk3) [reply packages/core/src/codewhispererChat/commands/registerCommands.ts]: why not?

File: packages/core/src/codewhispererChat/controllers/chat/controller.ts [hunk 3]
```diff
@@ -394,11 +394,23 @@ export class ChatController {
 
                 const prompt = this.promptGenerator.generateForContextMenuCommand(command)
 
-                this.messenger.sendEditorContextCommandMessage(
-                    command.type,
-                    context?.focusAreaContext?.codeBlock ?? '',
-                    triggerID
-                )
+                if (command.type === 'aws.amazonq.explainIssue' || command.type === 'aws.amazonq.fixIssue') {
```
Comment (justinmk3) [inline packages/core/src/codewhispererChat/controllers/chat/controller.ts:397]: off topic: command.name or command.id would be a more appropriate name.
```suggestion
                if (command.id === 'aws.amazonq.explainIssue' || command.id === 'aws.amazonq.fixIssue') {
```

**Review by justinmk3 — APPROVED** [review]

**Review by AnirudhKonduru — APPROVED** [review]

**Pull request merged**